go/arrow/csv/reader.go

```diff
@@ -58,7 +64,14 @@ type Reader struct {
 func NewReader(r io.Reader, schema *arrow.Schema, opts ...Option) *Reader {
 	validate(schema)
 
-	rr := &Reader{r: csv.NewReader(r), schema: schema, refs: 1, chunk: 1}
+	rr := &Reader{
+		r:                csv.NewReader(r),
+		schema:           schema,
+		refs:             1,
+		chunk:            1,
+		stringsCanBeNull: false,
+		nullValues:       []string{"", "NULL", "null"},
```

Comment:
I would use a `map[string]struct{}` instead.
but I'll send a separate PR to address it.

Reply:
not familiar with go, but if that is a hash-map for lookups, then for short lists iteration is often times cheaper.